README.md

```diff
@@ -145,6 +141,24 @@ that allows use of all the functionality described here, but in the `napari` def
 viewer itself.
 
 
+## Performance Considerations
+* There are two different ways for `aicsimageio` to load imaging data from a file,
+either delayed or immediately in memory. Internally, `aicsimageio` determines how to
+load the image data based off of which function you ran and if there is a
+`distributed.Client` available in the current Python process. The only situation in
+which imaging data will be read in a non-delayed fashion is when using `AICSImage.data`
+or `Reader.data` while also not having a `distributed.Client` available in the current
+Python process. This means that for all other situations and functions,
+(`AICSImage.dask_dask`, `AICSImage.get_image_data`, `AICSImage.get_image_dask_data`,
+etc.), the imaging data is retrieved in a delayed fashion. We do this to optimize read
+performance for the majority of situations we encounter as well as additionally
+supporting file reading of any size imaging file. The benefits of this strategy can be
+seen in our [benchmarks](https://allencellmodeling.github.io/aicsimageio/benchmarks.html).
+* When using the `dask_data` array, it is important to know when to `compute` or
+`persist` data and when to keep chaining computation.
+[Here is a good rundown on the trade offs.](https://stackoverflow.com/questions/41806850/dask-difference-between-client-persist-and-client-compute#answer-41807160)
+
```

Comment:
Being a naive end-user, I want to know which read functions to use under what conditions / use cases.  

If you know your image will fit into memory and is local, then AICSImage.data is generally sufficient and optimal.   
If your image is very large, and you need to read it piece by piece, then get_image_data will hand you a numpy array, and get_image_dask_data will hand you a delayed dask array which you then have to .compute() to do the final loading.
If you have a very large image but know you need to iterate over the entire image you should use.... 
You can expect slower-than-optimal performance with get_image_data if....

Is there documentation like that anywhere?

Reply:
If you think people will already figure that out from the existing docs, then I'm cool with it.